test/unit/rules/block-indentation-test.js

```diff
@@ -334,6 +334,110 @@ generateRuleTests({
         filePath: 'layout.js',
       },
     },
+    {
+      template: [
+        '<template>',
+        '  <div class="parent">',
+        '    <div class="child"></div>',
+        '  </div>',
+        '</template>',
+      ].join('\n'),
+      meta: {
+        filePath: 'layout.gjs',
+      },
+    },
+    {
+      template: [
+        '<template><div class="parent">',
+        '  <div class="child"></div>',
+        '</div></template>',
+      ].join('\n'),
+      meta: {
+        filePath: 'layout.gjs',
+      },
+    },
+    {
+      template: [
+        'const MyComponent = <template>',
+        '  <div class="parent">',
+        '    <div class="child"></div>',
+        '  </div>',
+        '</template>',
+      ].join('\n'),
+      meta: {
+        filePath: 'layout.gjs',
+      },
+    },
+    {
+      template: [
+        'function fetchComponent() {',
+        '  return <template>',
+        '    <div class="parent">',
```

Comment:
This is example of the `columnOffset` (= indentation of the line that contains `<template`, in this case 2 spaces before `return`) being necessary to indent this at level 2 + 2 = 4.